source/m2mnsdlinterface.cpp

```diff
@@ -228,15 +232,43 @@ bool M2MNsdlInterface::create_bootstrap_resource(sn_nsdl_addr_s *address, const
         memory_free(_endpoint->endpoint_name_ptr);
     }
 
-    _endpoint->endpoint_name_ptr = alloc_string_copy((uint8_t*)bootstrap_endpoint_name.c_str(), bootstrap_endpoint_name.length());
+    _endpoint->endpoint_name_ptr = alloc_string_copy((uint8_t*)bootstrap_endpoint_name.c_str(),
+                                                     bootstrap_endpoint_name.length());
     _endpoint->endpoint_name_len = bootstrap_endpoint_name.length();
     if(_bootstrap_id == 0) {
-        _bootstrap_id = sn_nsdl_oma_bootstrap(_nsdl_handle,
-                                               address,
-                                               _endpoint,
-                                               &bootstrap_endpoint);
-        tr_debug("M2MNsdlInterface::create_bootstrap_resource - _bootstrap_id %d", _bootstrap_id);
+        // Take copy of the address, uri_query_parameters() will modify the source buffer
+        bool msg_sent = false;
+        if (_server_address) {
+            char *address_copy = M2MBase::alloc_string_copy(_server_address);
+            char* query = query_string(_server_address);
+            if (query != NULL) {
```

Comment:
Fixed in https://github.com/ARMmbed/mbed-client/pull/405/commits/46c81590a5b3b03508f9fce5e2e180fd6bf7c915